index.rst

```diff
@@ -199,23 +223,55 @@ This class is a subclass of ``dmlc::Parameter``.
 
 Node
 ----
-Each node is represented as a JSON array of a fixed size, each element being
-determined by the ``node_schema`` field of XGBoostModel_.
+We may choose one of the two subclasses for this placeholder:
+
+* LeafNode_: leaf node (no child node, real output)
+* TestNode_: non-leaf node (two child nodes, test condition)
+
+The subclasses are distinguished by the boolean value of ``is_leaf`` field.
+
+LeafNode
+--------
+Each leaf node is represented as a JSON array of a fixed size, each element
+storing the following fields:
 
 .. parsed-literal::
 
   [
-    *integer* (parent_id),
+    true (is_leaf),
+    *floating-point* (leaf_output)
+  ]
+
+The ``is_leaf`` field (first entry) shall be set to ``true`` for all leaf nodes.
```

Comment:
but why? JSON parser is able to distinguish an array from a float - arrays begin from `[`

Reply:
I wanted to leave room for possible future addition of fields.

Reply:
Actually, never mind. I’ll follow your advice here. If any extra addition is merited, it can be added NodeStat.